gpbft/gpbft.go

```diff
@@ -890,39 +891,63 @@ func newQuorumState(powerTable PowerTable) *quorumState {
 	}
 }
 
-// Receives a new chain from a sender.
+// Receives a chain from a sender.
+// Ignores any subsequent value from a sender from which a value has already been received.
 func (q *quorumState) Receive(sender ActorID, value ECChain, signature []byte) {
-	senderPower, _ := q.powerTable.Get(sender)
+	senderPower, ok := q.receiveSender(sender)
+	if !ok {
+		return
+	}
+	q.receiveInner(sender, value, senderPower, signature)
+}
+
+// Receives each prefix of a chain as a distinct value from a sender.
+// Note that this method does not store signatures, so it is not possible later to
+// create an aggregate for these prefixes.
+// This is intended for use in the QUALITY phase.
+// Ignores any subsequent values from a sender from which a value has already been received.
+func (q *quorumState) ReceiveEachPrefix(sender ActorID, values ECChain) {
```

Comment:
Did you mean to use this? Or is this for a future patch?

Reply:
Thanks, good catch. A call to this replaces the equivalent code in the QUALITY_PHASE case in `receieveOne`.